playwright/tests/settings/settings.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { expect } from "@playwright/test";
+
+import { test } from "../../fixtures";
+import {
+  gitOwners,
+  gitRepos,
+  linkedRepos,
+  providerAppInstallURL,
+} from "../../mocks";
+import {
+  SliceMachineError,
+  UnauthenticatedError,
+  UnauthorizedError,
+} from "../../utils";
+
+test.run()(
+  "I can install the GitHub app",
+  async ({ procedures, settingsPage }) => {
+    procedures.mock("git.fetchLinkedRepos", () => [], { execute: false });
+    procedures.mock("git.fetchOwners", () => [], { execute: false });
+
+    await settingsPage.goto();
+
+    procedures.mock(
+      "git.getProviderAppInstallURL",
+      () => providerAppInstallURL,
+      { execute: false },
+    );
+
+    const gitHubPagePromise = settingsPage.page.context().waitForEvent("page");
+    await settingsPage.installGitHubButton.click();
+    const gitHubPage = await gitHubPagePromise;
+    await gitHubPage.waitForLoadState();
```

Comment:
I don't really understand the goal here. For me, we should try to test as a user can test. Can you not expect an element to be visible or not here? What the user should see after the click, that's waht we want to expect